packages/flutter_tools/lib/src/base/user_messages.dart

```diff
@@ -18,11 +18,22 @@ class UserMessages {
       'Channel ${channel ?? 'unknown'}, ${version ?? 'Unknown'}, on $os, locale $locale';
   String flutterVersion(String version, String channel, String flutterRoot) =>
       'Flutter version $version on channel $channel at $flutterRoot';
+  String get flutterUnknownChannel =>
+    'Currently on an unknown channel. Run `flutter channel` to switch to an official channel.\n'
```

Comment:
I feel like these messages are implying that the user's SDK installation is broken, which may not be the case. And from experience, we've found that not-green validations in the Flutter doctor can cause people a lot of anxiety. What about updating all of these to make it clear that if their current state is intentional, they are fine to continue?

Reply:
That is fine, but I prefer to avoid users confused by seeing the error messages from upgrade and have to be pointed out the "potential" causes, by other users comparing the doctor output.

Reply:
How about "It is fine to continue if this is intentional, however it is recommended to use "git" directly as the tool won't be able to perform upgrades_(and inform about new versions, wherever applicable)_ at this state".

Reply:
Added a message when FlutterValidator fails stating the above.